client/client_test.go

```diff
@@ -6641,6 +6649,8 @@ func testExportAttestations(t *testing.T, sb integration.Sandbox) {
 }
 
 func testAttestationDefaultSubject(t *testing.T, sb integration.Sandbox) {
+	integration.SkipIfDockerdMoby(t, sb, "direct push")
+	integration.SkipIfDockerdSnapshotter(t, sb, "number of platforms does not match references 1 2")
```

Comment:
https://github.com/crazy-max/buildkit/actions/runs/3257051098/jobs/5348086746#step:8:1111

```
=== CONT  TestIntegration/TestAttestationDefaultSubject/worker=dockerd-containerd
    client_test.go:6745: 
        	Error Trace:	client_test.go:6745
        	            				run.go:86
        	            				run.go:189
        	Error:      	Received unexpected error:
        	            	number of platforms does not match references 1 2
        	            	github.com/moby/buildkit/util/stack.Enable
        	            		/src/util/stack/stack.go:77
        	            	github.com/moby/buildkit/util/grpcerrors.FromGRPC
        	            		/src/util/grpcerrors/grpcerrors.go:196
        	            	github.com/moby/buildkit/util/grpcerrors.UnaryClientInterceptor
        	            		/src/util/grpcerrors/intercept.go:41
        	            	google.golang.org/grpc.(*ClientConn).Invoke
        	            		/src/vendor/google.golang.org/grpc/call.go:35
        	            	github.com/moby/buildkit/api/services/control.(*controlClient).Solve
        	            		/src/api/services/control/control.pb.go:1535
        	            	github.com/moby/buildkit/client.(*Client).solve.func2
        	            		/src/client/solve.go:231
        	            	golang.org/x/sync/errgroup.(*Group).Go.func1
        	            		/src/vendor/golang.org/x/sync/errgroup/errgroup.go:57
        	            	runtime.goexit
        	            		/usr/local/go/src/runtime/asm_amd64.s:1594
        	            	failed to solve
        	            	github.com/moby/buildkit/client.(*Client).solve.func2
        	            		/src/client/solve.go:244
        	            	golang.org/x/sync/errgroup.(*Group).Go.func1
        	            		/src/vendor/golang.org/x/sync/errgroup/errgroup.go:57
        	            	runtime.goexit
        	            		/usr/local/go/src/runtime/asm_amd64.s:1594
        	Test:       	TestIntegration/TestAttestationDefaultSubject/worker=dockerd-containerd
```

[TestAttestationDefaultSubject.log](https://github.com/moby/buildkit/files/9793226/TestAttestationDefaultSubject.log)

Reply:
:eyes: uh oh, something seems actually wrong there...

That error message is usually a signal that something has gone *quite* wrong - the number of refs stop matching up with the number of things that use them.

I'm fairly sure this isn't correct anymore, the error message is also changed recently to be slightly more indicative of the actual problem: https://github.com/moby/buildkit/blob/master/exporter/containerimage/writer.go#L118.

Reply:
Yes the `snapshot` adapter does not seem to match next v0.11 but v0.10 atm: https://github.com/rumpl/moby/blob/c8d/builder/builder-next/adapters/snapshot/snapshot.go